src/odemis/driver/test/xt_client_test.py

```diff
@@ -956,6 +955,151 @@ def test_beamlet_index_info(self):
+
+        # Loop over scan modes twice to make sure also the switch from fib to ebeam work properly
+        for _ in range(2):
+            for scanner_mode in scanner_modes:
+                # Switch active scanner mode
+                self.detector.scanner.value = scanner_mode
+                self.assertEqual(self.detector.scanner.value, scanner_mode)
+
+        # Check if mode remains unchanged if any non existing mode is tried to be set
+        with self.assertRaises(IndexError):
+            self.detector.scanner.value = "non existing mode"
+        # The mode should be equal to the last set correct mode.
+        self.assertEqual(self.detector.scanner.value, scanner_mode)
+
+    def test_subscribing_while_changing_scanner(self):
+        """
+        Subscribes to the dataflow then switches between both scanner types and then checks if an images is received
+        while subscribing to that type of scanner (FIB <--> ebeam)
+        """
+        self.images_received = 0
+        self.detector.data.subscribe(self._callback_counter)
+
+        scanner_modes = {self.scanner.name: self.scanner,
+                         self.fib_scanner.name: self.fib_scanner}
+
+        # Loop over scan modes twice to make sure also the switch from fib to ebeam work properly.
+        for _ in range(2):
+            for scanner_mode in scanner_modes:
+                # Switch active scanner mode
+                self.detector.scanner.value = scanner_mode
+                # Set image counter to zero to check that new images are received after switching
+                self.images_received = 0
+                self.assertEqual(self.detector.scanner.value, scanner_mode)
+                self.assertIsInstance(self.detector._scanner, type(scanner_modes[scanner_mode]))
+                time.sleep(2)
+                self.assertGreaterEqual(self.images_received, 1)
+
+        self.detector.data.unsubscribe(self._callback_counter)  # Unsubscribe so to stop refreshing the image
+
+    def test_acquire_FIB_image(self):
+        """Test acquiring an image from the FIB/ion channel."""
+        # Switch to ion mode
+        self.detector.scanner.value = self.fib_scanner.name
+        image = self.detector.data.get()
+        self.assertEqual(image.ndim, 2)
+        # Check that image size is at least 200*200 pixels
+        self.assertGreaterEqual(image.shape[0], 200)
+        self.assertGreaterEqual(image.shape[1], 200)
+
+    def test_acquire_ebeam_image(self):
+        """Test acquiring an image using the Detector."""
+        # Switch to electron mode
+        self.detector.scanner.value = self.scanner.name
+        init_dwell_time = self.scanner.dwellTime.value
+        self.scanner.dwellTime.value = 25e-9  # s
+        expected_duration = self._compute_expected_duration()
+        start = time.time()
+        im = self.detector.data.get()
+        duration = time.time() - start
+        # self.assertEqual(im.shape, self.scanner.resolution.value[::-1])
```

Comment:
why is this commented?

Reply:
No idea, I will test this again on Helios but I shouldn't have been commented.